pkg/abaputils/abaputils.go

```diff
@@ -142,15 +142,20 @@ func HandleHTTPError(resp *http.Response, err error, message string, connectionD
 		var abapErrorResponse AbapError
 		bodyText, readError := ioutil.ReadAll(resp.Body)
 		if readError != nil {
-			return readError
+			return err
 		}
 		var abapResp map[string]*json.RawMessage
-		json.Unmarshal(bodyText, &abapResp)
-		json.Unmarshal(*abapResp["error"], &abapErrorResponse)
-		if (AbapError{}) != abapErrorResponse {
-			log.Entry().WithField("ErrorCode", abapErrorResponse.Code).Error(abapErrorResponse.Message.Value)
-			abapError := errors.New(abapErrorResponse.Code + " - " + abapErrorResponse.Message.Value)
-			err = errors.Wrap(abapError, err.Error())
+		errUnmarshal := json.Unmarshal(bodyText, &abapResp)
+		if errUnmarshal != nil {
+			return err
+		}
+		if _, ok := abapResp["error"]; ok {
+			json.Unmarshal(*abapResp["error"], &abapErrorResponse)
+			if (AbapError{}) != abapErrorResponse {
+				log.Entry().WithField("ErrorCode", abapErrorResponse.Code).Error(abapErrorResponse.Message.Value)
+				abapError := errors.New(abapErrorResponse.Code + " - " + abapErrorResponse.Message.Value)
+				err = errors.Wrap(abapError, err.Error())
+			}
 		}
 		resp.Body.Close()
```

Comment:
This line should go up before any returns and be in a defer().

Reply:
good point